Source/WebCore/loader/TextResourceDecoder.cpp

```diff
@@ -36,35 +36,25 @@
 #include <wtf/text/MakeString.h>
 #include <wtf/text/ParsingUtilities.h>
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-
 namespace WebCore {
 
 using namespace HTMLNames;
 
-static constexpr bool bytesEqual(const uint8_t* p, uint8_t b)
-{
-    return *p == b;
-}
-
-template<typename... T>
-static constexpr bool bytesEqual(const uint8_t* p, uint8_t b, T... bs)
-{
-    return *p == b && bytesEqual(p + 1, bs...);
-}
+constexpr std::array<uint8_t, 3> bom = { 0xEF, 0xBB, 0xBF };
 
 // You might think we should put these find functions elsewhere, perhaps with the
 // similar functions that operate on UChar, but arguably only the decoder has
 // a reason to process strings of char rather than UChar.
 
-static int find(const uint8_t* subject, size_t subjectLength, const char* target)
+static size_t find(std::span<const uint8_t> subject, std::span<const uint8_t> target)
 {
-    size_t targetLength = strlen(target);
-    if (targetLength > subjectLength)
-        return -1;
-    for (size_t i = 0; i <= subjectLength - targetLength; ++i) {
+    if (target.size() > subject.size())
+        return notFound;
+
+    size_t sizeDifference = subject.size() - target.size();
+    for (size_t i = 0; i <= sizeDifference; ++i) {
```

Comment:
Not that it’s changing here, but I’m really surprised that this is `<=` rather than `<`.